CorePluginFramework/src/au/gov/asd/tac/constellation/plugins/gui/FileInputPane.java

```diff
@@ -83,41 +89,59 @@ public FileInputPane(final PluginParameter<FileParameterValue> parameter, final
-                    if (openFile != null) {
-                        files.add(openFile);
-                    }
+                case OPEN_OBSCURED:
+                    dialogFuture = FileChooser.openOpenDialog(getFileChooser(parameter, "Open")).thenAccept(optionalFile -> optionalFile.ifPresent(openFile -> {
+                        if (openFile != null) {
+                            files.add(openFile);
+                        }
+                    }));
                     break;
                 case OPEN_MULTIPLE:
-                    final List<File> multipleOpenFiles = fileChooser.showOpenMultipleDialog(getScene().getWindow());
-                    if (multipleOpenFiles != null) {
-                        files.addAll(multipleOpenFiles);
-                    }
+                case OPEN_MULTIPLE_OBSCURED:
+                    dialogFuture = FileChooser.openMultiDialog(getFileChooser(parameter, "Open Multiple")).thenAccept(optionalFile -> optionalFile.ifPresent(openFiles -> {
+                        if (openFiles != null) {
+                            files.addAll(openFiles);
+                        }
+                    }));
                     break;
                 case SAVE:
-                    final File saveFile = fileChooser.showSaveDialog(getScene().getWindow());
-                    if (saveFile != null) {
-                        files.add(saveFile);
-                    }
+                case SAVE_OBSCURED:
+                    dialogFuture = FileChooser.openSaveDialog(getFileChooser(parameter, "Save")).thenAccept(optionalFile -> optionalFile.ifPresent(saveFile -> {
+                        if (saveFile != null) {
+                            
+                            //Save files may have been typed by the user and an extension may not have been specified.
+                            final String fnam = saveFile.getAbsolutePath();
+
+                            String expectedExtension = FileParameterType.getFileFilters(parameter).getExtensions().get(0);
+                            if (!fnam.toLowerCase().endsWith(expectedExtension)) {
+                                saveFile = new File(fnam + expectedExtension);
+                            }
+                            files.add(saveFile);
+                        }
+                    }));
                     break;
                 default:
+                    dialogFuture = null;
                     LOGGER.log(Level.FINE, "ignoring file selection type {0}.", paramaterValue.getKind());
                     break;
             }
-
-            if (!files.isEmpty()) {
+            
+            // As the dialog windows are completed on another thread 
+            // the execution of this method must wait until the thread has finnished executing.
+            if (dialogFuture != null){
+                try {
+                    dialogFuture.get();
+                } catch (InterruptedException | ExecutionException ex) {
```

Comment:
final and should split this into two and interrupt the thread on the interrupted exception